pagopa/pn-frontend-e2e
# PN-9306-e-2-e-destinatario-recapiti-legali-modifica-pec

## Short description
There is a missing step in test related to the **PN-9306** as you can see in the [confluence page](https://pagopa.atlassian.net/wiki/spaces/PN/pages/868778282/Mittente+-+Mapping+test-scenari)

## List of changes proposed in this pull request
- refactor some methods in `BasePage` file
- added `clearWebElementField` in `BasePage`, this will help to erase the content inside the input of any webelement because the `clear()` of `WebElement` isn't working properly with custom React components.
- refactor some methods and webelements in `RecapitiDestinatarioPage` file and added 3 new methods: `verificaPecModificabile(), verificaPecNonModificabile(), clicksuAnnulla()`
- refactor some methods and webelements in `HeaderPFSection` file
- added `@recapitiPG` tag aswell in `Hooks` file, in order to clean all contacts after the test finished
- removed unnecessary steps in `aggiungiPECPF()` and `aggiungiPECPG`
- refactor feature steps in `RecapitiPersonaFisicaTest` and add a new feature step: 
`And Si clicca sul bottone annulla per annullare la modifica della PEC e si verifica che non si possa modificare la PEC`
- adjusted all cucumber feature files with related feature steps


## How to test
Simply run `mvn clean test -Dcucumber.filter.tags=@TA_modificaPECPF` and verify if the test works fine.
